### DISCUSSION TITLE
TextAdept doesn't work with objeck-lsp

### DISCUSSION THREAD

**Discussion by ghost** [orbitalquark/textadept#388] [Q&A]
Objeck-lsp recently added support for operating over stdio so I tried to use it with TextAdept. This is my config:

```
lexer.detect_extensions.obs='objeck'

local lsp = require('lsp')

lsp.server_commands.objeck = 'obr C:/objeck-lsp/objeck_lsp.obe C:/objeck-lsp/objk_apis.json stdio'
```

Note:

This is objeck-lsp: https://github.com/objeck/objeck-lsp

As you can see I extracted it to C drive. You could extract it to anywhere you want, but you have to modify the paths accordingly.

You must have Objeck VM installed and added to PATH:

https://github.com/objeck/objeck-lang/releases

For example, I extracted to zip release into C drive so I have to put `C:/objeck-lang/bin` to PATH.

If you are on Windows like me, you have to put `C:/objeck-lang/lib/sdl` to PATH too.

You must have these environment variables set:

OBJECK_LIB_PATH set to `C:/objeck-lang/lib` and OBJECK_STDIO set to `binary`.

If you extracted to Objeck VM to another location, you have to modify the paths according to your setup.

After you done setting up everything, you are ready to use objeck-lsp with TextAdept. The problem is, it doesn't work. The server started correctly but there is no code completion at all. Please have a look @orbitalquark. Thank you.

**Comment by orbitalquark** [comment]
Please temporarily add `require('lsp').log_rpc = true` to your *~/.textadept/init.lua*, (re)start Textadept, start the language server, and attempt code completion. Then go to Tools > Language Server > Show Log and post the contents here.

**Reply by ghost** [reply]
The server crashed. I will report to objeck-lsp.

**Comment by objeck** [comment]
What is the binary output flag added?

**Reply by ghost** [reply]
Do you mean the environment variable OBJECK_STDIO? It's set to `binary` as the guide for Sublime said. The full command used is:

`obr C:/objeck-lsp/objeck_lsp.obe C:/objeck-lsp/objk_apis.json stdio`

**Reply by objeck** [reply]
Yes, thanks I will look at the issue this week
